### ISSUE TITLE
Bug: @typescript-eslint/no-unused-vars false positive for underscore destructure

### ISSUE THREAD

**Issue by stevage** [typescript-eslint/typescript-eslint#7162]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.0.4&sourceType=module&code=MYewdgzgLgBAbgQwDYRgXhgbUwcgII4A0MAjALrG4BCRMATGWQLABQrAZiAE4wAUokWJgD6xREjIwQ7eMggBKGAG9WMGAIggkAUwB0SEAHNe4+a1YBfIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYkACAXAngA4oDG0AlobgLQrzkB2uA9AwPbWwOzKIAm1AG4BDaKgyRE0aG2iQAviHlA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
const vals = [['A', 1], ['B', 2]]

for (const [_, val] of vals) {
  console.log(val)

}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-unused-vars": "error",
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

No error. "_" in a destructured array being the convention for indicating that a variable won't be used.

### Actual Result

Error @typescript-eslint/no-unused-vars

**Comment by Josh-Cena** [comment]
[You need to specify a `varsIgnorePattern`.](https://typescript-eslint.io/play/#ts=5.0.4&sourceType=module&code=MYewdgzgLgBAbgQwDYRgXhgbUwcgII4A0MAjALrG4BCRMATGWQLABQrAZiAE4wAUokWJgD6xREjIwQ7eMggBKGAG9WMGAIggkAUwB0SEAHNe4+a1YBfIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYkACAXAngA4oDG0AlobgLQrzkB2uA9AwPbWwOzKIAm1AG4BDaKgwBtcBBmRE0aG2iQANNJmZ1GqCLEBJAObtoiAArDcueQ3RQA+pC0QAvloC6r50A&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA)

[This is the same behavior as the base rule, because there's no TS-specific syntax here.](https://eslint.org/play/#eyJ0ZXh0IjoiY29uc3QgdmFscyA9IFtbJ0EnLCAxXSwgWydCJywgMl1dXG5cbmZvciAoY29uc3QgW18sIHZhbF0gb2YgdmFscykge1xuICBjb25zb2xlLmxvZyh2YWwpXG5cbn1cbiIsIm9wdGlvbnMiOnsicGFyc2VyT3B0aW9ucyI6eyJlY21hVmVyc2lvbiI6ImxhdGVzdCIsInNvdXJjZVR5cGUiOiJtb2R1bGUiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsibm8tdW51c2VkLXZhcnMiOlsiZXJyb3IiLHsidmFyc0lnbm9yZVBhdHRlcm4iOiJfIn1dfSwiZW52Ijp7ImJyb3dzZXIiOnRydWV9fX0=)

For array destructuring, you should omit the variable altogether:

```ts
const vals = [['A', 1], ['B', 2]]

for (const [, val] of vals) {
  console.log(val)

}
```

**Comment by stevage** [comment]
Yes, that workaround works in the specific case of one placeholder variable.

**Comment by bradzacher** [comment]
Array destructuring variables are completely optional - supplying a name is optional and usually done for documentation purposes only.

You could rewrite your code to
```ts
const [, val] of vals
```

And this would allow you to scale to an infinite number of unused destructuring elements.

Otherwise an ignore pattern like `^_` is very common and ignores all underscore-prefixed names.
